s3/src/main/scala/geotrellis/spark/io/s3/AmazonS3Client.scala

```diff
@@ -20,7 +20,7 @@ import com.amazonaws.ClientConfiguration
 import com.amazonaws.auth._
 import com.amazonaws.services.s3.{AmazonS3Client => AWSAmazonS3Client}
 import com.amazonaws.services.s3.model._
-import org.apache.commons.io.IOUtils
+import sun.misc.IOUtils
```

Comment:
Does this work with both Oracle and OpenJDK versions of the JRE?

Can't help to worry we're [taking a risk](https://blog.dripstat.com/removal-of-sun-misc-unsafe-a-disaster-in-the-making/) using this API. That said, if it provides measurable benefits and if properly documented (and what to do if it's ever removed), I'll stop worrying. ;-)

Reply:
@metasim yes, [works for both](https://github.com/locationtech/geotrellis/blob/master/.travis.yml#L27-L29)

Reply:
I think I'll revert this change back: [the sun.misc package is internal to Sun/Oracle and should not be used as they are not publicly supported, and not portable across all JVMs](http://www.oracle.com/technetwork/java/faq-sun-packages-142232.html).

Reply:
I think erring on the side of caution is best. If there's a bit performance difference, maybe we could implement something using java.nio? If so, happy to help.

Reply:
well, it's a bit faster, don't think it's too significant at the moment, but it would be good to remember that we're using a not optimal function to convert input stream into a byte array.